backend/backend-gen/src/main/java/eu/socialedge/hermes/backend/gen/SchedulePdfGenerator.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Hermes - The Municipal Transport Timetable System
+ * Copyright (c) 2016-2017 SocialEdge
+ * <p>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+package eu.socialedge.hermes.backend.gen;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import eu.socialedge.hermes.backend.transit.domain.service.Line;
+
+import eu.socialedge.hermes.backend.gen.data.StationScheduleTemplate;
+import eu.socialedge.hermes.backend.schedule.domain.Schedule;
+import eu.socialedge.hermes.backend.transit.domain.infra.Station;
+import lombok.val;
+
+public class SchedulePdfGenerator {
```

Comment:
This sounds like a `service`. Can we use something like `PdfScheduleSerializationService`?

Reply:
Yep